crates/repository/src/lib.rs

```diff
@@ -139,26 +140,19 @@ impl Repository {
         target: &str,
         passthrough_arguments: &[String],
     ) -> Result<()> {
-        let mut shell_command = String::new();
-
-        if target == "nao" {
-            shell_command += &format!(
-                ". {} && ",
-                self.root
-                    .join(format!(
-                        "naosdk/{SDK_VERSION}/environment-setup-corei7-64-aldebaran-linux"
-                    ))
-                    .display()
-            );
-        }
-
         let cargo_command = format!("cargo {action} ")
             + format!("--profile {profile} ").as_str()
             + if workspace {
                 "--workspace --all-features --all-targets ".to_string()
             } else {
+                let use_docker = target == "nao" && OS_IS_NOT_LINUX;
                 let manifest = format!("crates/hulk_{target}/Cargo.toml");
-                format!("--manifest-path={} ", self.root.join(manifest).display())
+                let root = if use_docker {
+                    Path::new("/hulk")
+                } else {
+                    &self.root
+                };
+                format!("--manifest-path={} ", root.join(manifest).display())
```

Comment:
The entire control flow here is now very confusing and complex. In some future iteration, I can think of a fancy builder syntax using type-states leading to compiletime-typesafe construction of the respective commands...

Resolve this conversation if read, maybe open an issue :shrug: